src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -45,6 +48,18 @@ internal void Match(RouteContext context)
                 }
             }
 
+            // In addition to extracting parameter values from the URL, each route entry
+            // also knows which other parameters should be supplied with null values. These
+            // are parameters supplied by other route entries matching the same handler.
+            if (UnusedRouteParameterNames.Length > 0)
```

Comment:
You should double check that this handles cases with optional parameters (if Blazor supports them) like `{a}/{b?}`. Depending on the implementation it's possible for you to get `b` omitted, or `b => null` in the dictionary and it seems like we need `b => null`.

Reply:
It’s a good thought, but Blazor routes don’t yet support optional parameters.

Reply:
OK - worth a comment somewhere then? It's really likely we'll try to converge on features at some point. If we have an issue tracking that then we could add the note there.

Reply:
Yes, commented here: https://github.com/aspnet/AspNetCore/issues/10759#issuecomment-513622534